src/histogram_equalization.jl

```diff
@@ -0,0 +1,217 @@
+        diff_img_nw = padded_img[0:end-2, 0:end-2] .- I
+        diff_img_ne = padded_img[0:end-2, 2:end] .- I
+        diff_img_sw = padded_img[2:end, 0:end-2] .- I
+        diff_img_se = padded_img[2:end, 2:end] .- I
+
+        # Exponential conduction coefficients
+        conduct_coeff_north = exp.(-(abs.(diff_img_north) ./ gradient_threshold) .^ 2)
+        conduct_coeff_east = exp.(-(abs.(diff_img_east) ./ gradient_threshold) .^ 2)
+        conduct_coeff_nw = exp.(-(abs.(diff_img_nw) ./ gradient_threshold) .^ 2)
+        conduct_coeff_ne = exp.(-(abs.(diff_img_ne) ./ gradient_threshold) .^ 2)
+        conduct_coeff_sw = exp.(-(abs.(diff_img_sw) ./ gradient_threshold) .^ 2)
+        conduct_coeff_se = exp.(-(abs.(diff_img_se) ./ gradient_threshold) .^ 2)
+
+
+        # Flux calculations
+        flux_north = conduct_coeff_north .* diff_img_north
+        flux_east = conduct_coeff_east .* diff_img_east
+        flux_nw = conduct_coeff_nw .* diff_img_nw
+        flux_ne = conduct_coeff_ne .* diff_img_ne
+        flux_sw = conduct_coeff_sw .* diff_img_sw
+        flux_se = conduct_coeff_se .* diff_img_se
+
+        # Back to regular 1-indexed arrays
+        flux_north_diff = flux_north[1:end-1, :] .- flux_north[2:end, :]
+        flux_east_diff = flux_east[:, 2:end] .- flux_east[:, 1:end-1]
+
+        # Discrete PDE solution
+        sum_ = (1 / (dd^2)) .* (flux_nw .+ flux_ne .+ flux_sw .+ flux_se)
+        I = I .+ diffusion_rate .* (flux_north_diff .- flux_north_diff .+ sum_)
+
+    end
+
+    return I
+end
+
+
+function imshow(img)
+    if typeof(img) <: BitMatrix
+        return Gray.(img)
+    end
+    Gray.(img ./ 255)
+end
+
+
+
+function adapthisteq(img::Matrix{T}, nbins=256, clip=0.01) where {T}
+    # Step 1: Normalize the image to [0, 1] based on its own min and max
+    image_min, image_max = minimum(img), maximum(img)
+    normalized_image = (img .- image_min) / (image_max - image_min)
+
+    # Step 2: Apply adaptive histogram equalization. equalize_adapthist handles the tiling to 1/8 of the image size (equivalent to 8x8 blocks in MATLAB)
+    equalized_image = sk_exposure.equalize_adapthist(
+        normalized_image,
+        clip_limit=clip,  # Equivalent to MATLAB's 'ClipLimit'
+        nbins=nbins         # Number of histogram bins. 255 is used to match the default in MATLAB script
+    )
+
+    # Step 3: Rescale the image back to the original range [image_min, image_max]
+    final_image = sk_exposure.rescale_intensity(equalized_image, in_range="image", out_range=(image_min, image_max))
+
+    # Convert back to the original data type if necessary
+    final_image = to_uint8(final_image)
+
+    return final_image
+end
+
+"""
+    getrbc_channels(img)
+
+Get the RBC (Red, Blue, and Green) channels of an image.
+
+# Arguments
+- `img`: The input image.
+
+# Returns
+An m x n x 3 array the Red, Blue, and Green channels of the input image.
+
+"""
+function getrbc_channels(img)
```

Comment:
```suggestion
function getrgb_channels(img)
```
I wasn't sure if it should be `rbg_channels` or if `rbc` has meaning as well.

Reply:
Nah, it's rgb! Thanks :)